state-sync/state-sync-v2/state-sync-driver/src/tests/driver.rs

```diff
@@ -372,3 +340,26 @@ async fn create_driver_for_tests(
         time_service,
     )
 }
+
+/// Waits for node auto bootstrapping by the driver
+async fn wait_for_auto_bootstrapping(validator_driver: DriverFactory, time_service: TimeService) {
+    // Create the driver client and a join handle that waits on auto bootstrapping
+    let driver_client = validator_driver.create_driver_client();
+    let auto_bootstrapping_handle = tokio::spawn(async move {
+        driver_client.notify_once_bootstrapped().await.unwrap();
+    });
+
+    // Spawn a task that continuously elapses time
+    tokio::spawn(async move {
+        time_service
+            .into_mock()
+            .advance_async(Duration::from_secs(
+                StateSyncDriverConfig::default().max_connection_deadline_secs,
+            ))
+            .await;
+        sleep(Duration::from_secs(1));
```

Comment:
If you're using `time_service` wouldn't you want to use `time_service`.sleep?

Reply:
The driver uses the time service. The `sleep` here is to make the tokio test thread/task sleep between time service ticks 😄 It's just to pace out the increments of the time service.

Reply:
you want to use an async sleep though?

Reply:
> you want to use an async sleep though?

You mean on the time service or in the task? Either way, it shouldn't matter, right? All the tests that use it are using multithreaded runtimes. Or, am I missing something else?

Reply:
> using multithreaded runtimes
That's not a reason for calling blocking functions in an async context man 😂 let's try to avoid that and make it a custom if no special reason..

Reply:
Test code. Being lazy. But, agreed 😄